boa_parser/src/parser/expression/primary/async_function_expression/mod.rs

```diff
@@ -133,7 +142,12 @@ where
             params_start_position,
         )?;
 
-        let function = AsyncFunction::new(name, params, body, has_binding_identifier);
+        let function = AsyncFunction::new(
+            if name.is_some() { name } else { self.name },
```

Comment:
`unwrap_or`? Though, this won't apply anymore if we go with the `FunctionName` suggestion.

Reply:
`unwrap_or` does not work because it takes `T`. `self.name` is also `Option<T>`. I'm not sure if there if a dedicated function on `Option` for this operation.

Reply:
Ah, should be `or` then.